### PR TITLE
DYN-10096: On package install engine_version checks are suppressed if compatibility_matrix check passes

### PR DESCRIPTION
**Pull request by ivaylo-matov**
### Purpose

This PR fixes [DYN-10069](https://autodesk.atlassian.net/browse/DYN-10069?atlOrigin=eyJpIjoiNTg0NWE1N2JlZDUzNDY4NDgwZTk0YWJjYjhiNmExNDYiLCJwIjoiaiJ9), where Clockwork installation on Dynamo 3.6.1 incorrectly triggers engine version warnings despite valid compatibility.

Root cause is in `PackageManagerClientViewModel.ExecutePackageDownload()`, which evaluates both `compatibility_matrix` and published `engine_version`. The engine version check can override valid compatibility results.

Fix:

- made `compatibility_matrix` the primary compatibility source
- only fall back to `engine_version` when compatibility is unknown or incompatible
- removed hardcoded Dynamo 4.x assumption in engine version logic
- generalized warning to any dependency with a lower major engine version than the current Dynamo runtime
- added unit test to ensure no warnings are raised when `compatibility_matrix` reports compatibility

### Declarations

Check these if you believe they are true

- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Release Notes

Fixes incorrect package compatibility warnings during installation when packages are marked compatible via the compatibility matrix.

### Reviewers

@DynamoDS/eidos
@jasonstratton

### FYIs

@johnpierson 
@dnenov

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 1]
```diff
@@ -965,60 +965,65 @@ internal async void ExecutePackageDownload(string name, PackageVersion package,
...
                 // Determine if there are any dependencies that have a newer dynamo version, (this includes the root package).
-                // We assume this means this package is compatibile with that dynamo version, but we should warn the user it
+                // We assume this means this package is compatible with that dynamo version, but we should warn the user it
                 // may not work with the current version of Dynamo.
 
-                //wrap in try catch as it's possible version info could be missing. If so, we install the package and log an error.
-                try
+                // If the compatibility matrix already indicates compatibility, skip the engine_version heuristic.
+                // Only perform engine_version based warnings when compatibility is not explicitly true.
+                if (compatible == false)
```
Comment (jasonstratton) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs]: Should this be if (compatible != true)? ... There are actually 3 states, true, false and null

File: test/DynamoCoreWpf2Tests/PackageManager/PackageManagerUITests.cs [hunk 2]
```diff
@@ -1626,6 +1626,82 @@ public void PackageManagerShowsWarningWhenDownloadingLegacyPackage()
                         MessageBoxButton.OKCancel,MessageBoxImage.Exclamation), Times.Exactly(1));
 
         }
+
+        [Test]
+        [Description("If the package compatibility matrix explicitly marks the package as compatible, do NOT show engine_version newer/older warnings even when engine_version differs.")]
+        public void PackageManagerDoesNotShowEngineVersionWarningWhenCompatibilityMatrixIsCompatible()
```
Comment (jasonstratton) [inline test/DynamoCoreWpf2Tests/PackageManager/PackageManagerUITests.cs:1632]: Tests the happy path, please add a test for the null case (no matrix → warning should still appear)

**Review by jasonstratton — APPROVED** [review]
Address the two comments and I think it's good. ... Just let me know when you're done in order to merge.

File: test/DynamoCoreWpf2Tests/PackageManager/PackageManagerUITests.cs (hunk 2 above)
Comment (ivaylo-matov) [reply test/DynamoCoreWpf2Tests/PackageManager/PackageManagerUITests.cs:1632]: `PackageManagerDoesNotShowEngineVersionWarningWhenCompatibilityMatrixIsCompatible` now added to test the `null` case

**Comment by ivaylo-matov** [comment]
> Address the two comments and I think it's good. ... Just let me know when you're done in order to merge.

Thanks for the review, @jasonstratton ! I've addressed both comments, the PR should be ready to merge.

**Comment by johnpierson** [comment]
After rebase, 3 failing tests pass. 
image

**Review by johnpierson — APPROVED** [review]
Tested locally and working great. Tested with Rhythm which is marked compatible with 4.0, but has an engine version of 2.17
